docs/authentication.rst

```diff
@@ -87,6 +87,36 @@ require authentication, no credentials are sent.
 .. autoclass:: requests_toolbelt.auth.guess.GuessAuth
 
 
+GuessAuthProxyAuth
+---------
+
+The :class:`~requests_toolbelt.auth.guess.GuessAuthProxyAuth` authentication class
+provides GuessAuth, while also automatically detecting whether to use basic auth
+or digest auth with a provided proxy.
+
+.. code-block:: python
+
+    import requests
+    from requests_toolbelt.auth.guess import GuessAuthProxyAuth
+
+    proxies = {
+        "http": "http://PROXYSERVER:PROXYPORT",
+        "https": "https://PROXYSERVER:PROXYPORT",
```

Comment:
The correct way to do this in requests is typically:

```
"https": "http://PROXYSERVER:PROXYPORT",
```

Because of issues surrounding how we talk to HTTPS sites over proxies run on HTTPS. Since we recommend the above in requests, our examples here should do the same.